docs/contributor-guide/developer-guide/technical-contribution-guide.md

```diff
@@ -18,13 +18,86 @@ Development tasks are managed in the [GitHub issues](https://github.com/civiform
 
 If you're just getting started, check out issues labeled with [Good First Issue](https://github.com/civiform/civiform/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). Also check out issues in the [next milestone](https://github.com/civiform/civiform/milestones?direction=asc\&sort=due\_date\&state=open) so you can work on the highest-priority tasks.
 
-### Pull requests
+### Git workflow
 
-When you're ready to submit your code, open a pull request with "Closes #X" to link the relevant issue.
+To start working on a change, first get the latest state from the github repository then create a new local development branch:
+
+```sh
+# Downloads any commits from the github repository that your local repository does not have.
+git fetch origin
+
+# Creates a new local branch `your_username/dev-branch-description`.
+# The repository file contents are set to the same as the github repository's main branch.
+git checkout -b your_username/dev-branch-description origin/main
```

Comment:
I think the vast majority of the time we're branching from main, so the `/main` suffix could probably be left out.